py/dml/dmlparse.py

```diff
@@ -2512,15 +2512,15 @@ def reserved(t):
 # Error handling
 @prod
 def error(t):
-    if t and t.value:
-        value = str(t.value)
-        raise ESYNTAX(DumpableSite(t.lexer.file_info, t.lexpos), value, None)
-    else:
+    if t is None:
         # raise ESYNTAX(...) would have made sense here, but there
         # is no clean way to retrieve the file (because t is
         # None). Therefore, push the error reporting responsibility to
         # whoever invokes the parser.
         raise UnexpectedEOF()
+    else:
+        value = str(t.value)
```

Comment:
What value did the previous `t.value` check have? You no longer perform it, so we must consider if it contributed anything.

Reply:
Likely no value. We should not preserve things out of superstition. Worst case is if t.value would be a false value that doesn't display nicely, and in this case we will move from an anonymous EOF error to an error that points at the correct token but presents that token as garbage. That's an improvement.

Reply:
I'm inclined to agree.